pkg/lightning/backend/kv/sql2kv.go

```diff
@@ -405,62 +410,82 @@ func (kvcodec *tableKVEncoder) Encode(
 	}
 	pairs := kvcodec.se.takeKvPairs()
 	kvcodec.recordCache = record[:0]
-	return kvPairs(pairs), nil
+	return pairs, nil
 }
 
-func (kvs kvPairs) Size() uint64 {
+func (kvs *KvPairs) Size() uint64 {
 	size := uint64(0)
-	for _, kv := range kvs {
+	for _, kv := range kvs.pairs {
 		size += uint64(len(kv.Key) + len(kv.Val))
 	}
 	return size
 }
 
-func (kvs kvPairs) ClassifyAndAppend(
+func (kvs *KvPairs) ClassifyAndAppend(
 	data *Rows,
 	dataChecksum *verification.KVChecksum,
 	indices *Rows,
 	indexChecksum *verification.KVChecksum,
 ) {
-	dataKVs := (*data).(kvPairs)
-	indexKVs := (*indices).(kvPairs)
+	dataKVs := (*data).(*KvPairs)
+	indexKVs := (*indices).(*KvPairs)
 
-	for _, kv := range kvs {
+	for _, kv := range kvs.pairs {
 		if kv.Key[tablecodec.TableSplitKeyLen+1] == 'r' {
-			dataKVs = append(dataKVs, kv)
+			dataKVs.pairs = append(dataKVs.pairs, kv)
 			dataChecksum.UpdateOne(kv)
 		} else {
-			indexKVs = append(indexKVs, kv)
+			indexKVs.pairs = append(indexKVs.pairs, kv)
 			indexChecksum.UpdateOne(kv)
 		}
 	}
 
+	// the related buf is shared, so we only need to set it into one of the kvs so it can be released
+	if kvs.bytesBuf != nil {
+		dataKVs.bytesBuf = kvs.bytesBuf
+		dataKVs.memBuf = kvs.memBuf
```

Comment:
Shall we set `kvs.bytesBuf = nil; kvs.memBuf = nil;` ? so that the reference of buf can be release.

Reply:
Done. In theory go GC should properly handle this but I still add the set nil action since it's not harmful and maybe help go GC better handle some situation because we depend on CGO to manage memory.